mp/src/game/client/momentum/mom_map_cache.cpp

```diff
@@ -374,6 +375,31 @@ void CMapCache::SetMapGamemode(const char *pMapName /* = nullptr*/)
     if (!pMapName)
         pMapName = MapName();
 
+    if (Q_strcmp(mom_gamemode_override.GetString(), "0") && Q_strcmp(mom_gamemode_override.GetString(), ""))
+    {
+        // converting the player's command to int in order to simplify the rest of the logic
+
+        GameMode_t pGameMode = g_pGameModeSystem->GetGameMode(mom_gamemode_override.GetInt())->GetType();
+
+        if (pGameMode == GAMEMODE_UNKNOWN)
+        {
+            pGameMode = g_pGameModeSystem->GetGameModeFromString(mom_gamemode_override.GetString())->GetType();
+        }
+
+        if (pGameMode != GAMEMODE_UNKNOWN)
+        {
+            if (m_pCurrentMapData && m_pCurrentMapData->m_eType == pGameMode || g_pGameModeSystem->GetGameModeFromString(pMapName)->GetType() == pGameMode)
+            {
+                mom_gamemode_override.Revert();
+                return;
+            }
+
+            g_pGameModeSystem->SetGameMode(pGameMode);
+            mom_gamemode_override.SetValue(pGameMode);
```

Comment:
Why is this being set here?

Reply:
If `mom_gamemode_override` is a string because of the player manually overriding it, it'll break the `GetBool` checks for the anti cheat and loading screen, with the only other workaround I could think of being checking via `GetBool` *and* `GetString` + `Q_strncmp` every single time

Reply:
It should never be a string, only int, meaning the change to parse the name of it from the prefix isn't needed. The docs page can have all the game mode ints on it so people can see what they need to set.

Reply:
Alright; in that case should I maybe have the cvar reference the docs in some way? Maybe having a list in the help text or a link to the docs page?

Reply:
No, would be too much hardcoding. It should just be inferred that if people want more info on how a convar or concommand works, they go to the docs site for it.